tensorflow/compiler/xla/service/gpu/horizontal_loop_fusion.h

```diff
@@ -93,6 +93,8 @@ namespace gpu {
 // output dims of the concatenate will be used as the kernel launch dims.
 // Instruction bitcasts can be used for Reshape2 and Reshape3 as long as the
 // outputs of Mul and Add are row-major.
+//
+// Note, reshapes are added only if useful to make the graph more readable.
```

Comment:
"only if needed for shapes to match" or similar? Because I understand reshapes are added not for being "useful", but because the shapes don't agree without them?